sirepo/sim_api/jupyterhublogin.py

```diff
@@ -49,6 +49,8 @@ def api_migrateJupyterhub():
 def api_redirectJupyterHub():
     is_new_user = _create_user()
     if not cfg.rs_jupyter_migrate or not is_new_user:
+        if is_new_user:
+            pkio.mkdir_parent(_user_dir())
```

Comment:
seems like this code should be in _create_user, since it is creating the user here. It's confusing to see so many conditions here. Rather test for migration in create_user.

Reply:
59f2d698 . I realized there was another case I didn't cover where the user dir wasn't created (migration enabled but the user says they don't want to migrate). I refactored so the user dir is always create. In the event of a migration the dir will be overwritten with the dir from the migration.